tests/unit/test_sonatadata.cpp

```diff
@@ -135,52 +135,57 @@ SCENARIO("Test SonataData class", "[SonataData][IOWriter]") {
     GIVEN("Spike data") {
         std::vector<double> spike_timestamps{0.3, 0.1, 0.2, 1.3, 0.7};
         std::vector<uint64_t> spike_node_ids{3, 5, 2, 3, 2};
-        std::unique_ptr<SonataData> sonata_spikes = std::make_unique<SonataData>("spikes",
-                                                                                 population_name,
-                                                                                 spike_timestamps,
-                                                                                 spike_node_ids);
-        WHEN("We write the spikes ordered by time") {
-            sonata_spikes->write_spikes_header("by_time");
+        uint64_t population_offset = 0;
+        std::unique_ptr<SonataData> sonata_spikes = std::make_unique<SonataData>("spikes");
+        std::unique_ptr<Population> sonata_population = std::make_unique<Population>(
+            population_name, population_offset, "by_time", spike_timestamps, spike_node_ids);
+        Population* pop_p = sonata_population.get();
+        WHEN("We add the population and write the spikes ordered by time") {
+            sonata_spikes->add_population(sonata_population);
+            sonata_spikes->write_spike_populations();
             THEN("We check that the spike nodes ids are ordered according to timestamps") {
-                const std::vector<uint64_t> node_ids = sonata_spikes->get_spike_node_ids();
+                const std::vector<uint64_t> node_ids = pop_p->get_spike_node_ids();
                 std::vector<uint64_t> compare = {5, 2, 3, 2, 3};
                 REQUIRE(node_ids == compare);
             }
             THEN("We check that the spike timestamps are in order") {
-                const std::vector<double> timestamps = sonata_spikes->get_spike_timestamps();
+                const std::vector<double> timestamps = pop_p->get_spike_timestamps();
                 std::vector<double> compare = {0.1, 0.2, 0.3, 0.7, 1.3};
                 REQUIRE(timestamps == compare);
             }
         }
         WHEN("We write the spikes ordered by id") {
-            sonata_spikes->write_spikes_header("by_id");
+            sonata_population->set_sorting("by_id");
```

Comment:
Not sure about the `WHEN`, `THEN` etc...is this a moved-from object? (similarly below)

Reply:
The test case is running without error so it seems that `WHEN` blocks are executed independently. `sonata_population` is not moved in this test case because it doesn't call `add_population(std::unique_ptr<Population> population)`